chainerx_cc/chainerx/cuda/cuda_device.h

```diff
@@ -47,14 +52,37 @@ class MemoryKeeper {
     std::queue<std::pair<cudaEvent_t, std::shared_ptr<void>>> queue_{};
 };
 
-}  // namespace cuda_internal
-
-class CudaDevice : public Device {
+class DeviceInternals {
 public:
-    ~CudaDevice() override;
+    DeviceInternals(const DeviceInternals&) = delete;
+    DeviceInternals(DeviceInternals&&) = delete;
+    DeviceInternals& operator=(const DeviceInternals&) = delete;
+    DeviceInternals& operator=(DeviceInternals&&) = delete;
+
+    explicit DeviceInternals(int index) : index_{index}, cublas_handle_{index}, cudnn_handle_{index} {}
+
+    cuda_internal::CublasHandle& cublas_handle() { return cublas_handle_; }
 
     cuda_internal::CudnnHandle& cudnn_handle() { return cudnn_handle_; }
 
+    cuda_internal::CudaConv& cuda_conv() { return cuda_conv_; }
+
+private:
+    int index_;
```

Comment:
Any reason to store the index?

Reply:
Thanks, it seems not to be needed. Removed in https://github.com/chainer/chainer/pull/6827.